### PR TITLE
Ks dependabotalertsapi

### PR DESCRIPTION
**Pull request by kylestubbs102**
migrated to add dependabot alerts to db, then added route and controller for api

### CODE REVIEW

File: app/controllers/api/dependabot_alert_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  # last line checks for ORG_ID NOT NAME,
+  def check_org_and_repo?
+    @repository = Repository.find_by(name: params[:repo_id])
+    @organization = Organization.find_by(name: params[:org_id])
+
+    !(@repository.nil? || 
+      @organization.nil? || 
+      @user.organizations.nil? ||
+      !@user.organizations.include?(@organization) ||
+      !@repository['organization_id'] == @organization['id'])
```
Comment (chrisyang365) [inline app/controllers/api/dependabot_alert_controller.rb]: Sorry forgot to mention but we'll be storing Repository name's like this `org_name/repo_name`, so you can simply just lookup the repo by repo name, which will just be the `org_id` and `repo_id` path parameters with a `/` between the two

File: app/models/dependabot_alert.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class DependabotAlert < ApplicationRecord
+    validates :name, presence: true
```
Comment (chrisyang365) [inline app/models/dependabot_alert.rb:3]: probably want to validate all fields

File: db/migrate/20220117034711_create_dependabot_alerts.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreateDependabotAlerts < ActiveRecord::Migration[6.1]
+  def change
+    create_table :dependabot_alerts do |t|
+      t.string :name
+      t.string :description
+      t.string :severity
+      t.string :status
```
Comment (chrisyang365) [inline db/migrate/20220117034711_create_dependabot_alerts.rb:7]: Forgot to mention but we'll also have to change the primary key for Dependabot Alerts to be a string, and call it `node_id`. I'm not exactly sure how to do this but should be achievable with some Googling ;)

Comment (kylestubbs102) [reply db/migrate/20220117034711_create_dependabot_alerts.rb:7]: Ok so have the name and the node_id, or just the node_id?

Comment (chrisyang365) [reply db/migrate/20220117034711_create_dependabot_alerts.rb:7]: yeah both. node_id is gonna be like a random hash of characters

**Pull request merged**